File: MYCM/core/persistence/PersistenceContext.cs
using System;

namespace core.persistence{
    /// <summary>
    /// Class that acts as a factory of factories for a certain persistence context
    /// </summary>
    public sealed class PersistenceContext{
        /// <summary>
        /// RepositoryFactory with the current persistence context repository factory
        /// </summary>
        private static RepositoryFactory contextRepositoryFactory;

        /// <summary>
        /// Returns all repositories available for the current persistence context
        /// </summary>
        /// <returns>RepositoryFactory with the available repositories</returns>
        public static RepositoryFactory repositories(){
            if(contextRepositoryFactory==null){
                RepositoryFactory repositoryFactory
                                        =(RepositoryFactory)Activator
                                                            .CreateInstance(Type
                                                                .GetType(Application
                                                                    .settings()
                                                                    .getPersistenceContext()));
                if(repositoryFactory==null)
                    throw new InvalidOperationException();
                contextRepositoryFactory=repositoryFactory;
            }
            return contextRepositoryFactory;
        }

        /// <summary>
        /// Hides default constructor
        /// </summary>
        private PersistenceContext(){}
    }
}

File: MYCA/myca/core/src/main/java/cdiomyc/core/persistence/PersistenceContext.java
package cdiomyc.core.persistence;

import cdiomyc.core.application.Application;

/**
 * Persistence Utility class that represents the factory of persistence repositories 
 * factories for a certain persistence context
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 */
public final class PersistenceContext {
    /**
     * Constant that represents the message that occurres if the current 
     * persistence context is not defined
     */
    private static final String PERSISTENCE_CONTEXT_NOT_DEFINED_MESSAGE
            ="The persistence context is not defined!";
    /**
     * RepositoryFactory with the current persistence context repository factory
     */
    private static RepositoryFactory contextRepositoryFactory;
    /**
     * Method that returns the repositories of the current persistence context
     * @return RepositoryFactory with the factory that allows the repositories 
     * retrieval of the current persistence context
     */
    public static RepositoryFactory repositories(){
        if(contextRepositoryFactory==null){
            try {
                return (RepositoryFactory)
                        Class.forName(Application.settings().getPersistenceContext()).newInstance();
            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException persistenceContextException) {
                throw new IllegalStateException(PERSISTENCE_CONTEXT_NOT_DEFINED_MESSAGE,persistenceContextException);
            }
        }
        return contextRepositoryFactory;
    }
    /**
     * Hides default constructor
     */
    private PersistenceContext(){}
}
